### PR TITLE
[ci] for autogenerating docs

### PR DESCRIPTION
**Pull request by nabokovas**
Comments were reworked there for easier autogenerating documentation.
No changes for the functions.
No description for LGBM_NetworkInitWithFunctions.

### CODE REVIEW

**Comment by StrikerRUS** [comment]
Awesome!

Great addition for #2059!

@nabokovas Please fix the conflict before I start the review.

@guolinke I suppose this PR should be merged **after** #2059, right?

File: include/LightGBM/c_api.h [hunk 1]
```diff
@@ -283,39 +296,45 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetGetFeatureNames(
...
-* \param type C_API_DTYPE_FLOAT32 or C_API_DTYPE_INT32
-* \return 0 when succeed, -1 when failure happens
+
+ * @fn LGBM_DatasetSetField
+ * @brief Set vector to a content in info.
+ *        Note: group and group only work for C_API_DTYPE_INT32,
+ *              label and weight only work for C_API_DTYPE_FLOAT32.
+ * @param handle An instance of dataset
+ * @param field_name Field name, can be label, weight, group, group_id
+ * @param field_data Pointer to vector
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `to vector` -> `to data vector`

File: include/LightGBM/c_api.h [hunk 2]
```diff
@@ -283,39 +296,45 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetGetFeatureNames(
...
+ * @param num_element Number of element in field_data
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `of elements`

File: include/LightGBM/c_api.h [hunk 3]
```diff
@@ -283,39 +296,45 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetGetFeatureNames(
...
+ * @param type C_API_DTYPE_FLOAT32 or C_API_DTYPE_INT32
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: ` * @param type Type of data pointer, can be C_API_DTYPE_INT8, C_API_DTYPE_INT32, C_API_DTYPE_FLOAT32 or C_API_DTYPE_FLOAT64`

File: include/LightGBM/c_api.h [hunk 4]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
-* \brief get info vector from dataset
-* \param handle a instance of data matrix
-* \param field_name field name
-* \param out_len used to set result length
-* \param out_ptr pointer to the result
-* \param out_type  C_API_DTYPE_FLOAT32 or C_API_DTYPE_INT32
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_DatasetGetField
+ * @brief Get info vector from dataset.
+ * @param handle An instance of data matrix
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `An instance of data matrix` -> `An instance of dataset`

File: include/LightGBM/c_api.h [hunk 5]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
+ * @param field_name Field name
+ * @param out_len Used to set result length
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `@param[out]`

File: include/LightGBM/c_api.h [hunk 6]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
+ * @param out_ptr Pointer to the result
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `@param[out]`

File: include/LightGBM/c_api.h [hunk 7]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
+ * @param out_type C_API_DTYPE_FLOAT32 or C_API_DTYPE_INT32
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: ` * @param[out] out_type Type of result pointer, can be C_API_DTYPE_INT8, C_API_DTYPE_INT32, C_API_DTYPE_FLOAT32 or C_API_DTYPE_FLOAT64`

File: include/LightGBM/c_api.h [hunk 8]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
                                            const void** out_ptr,
                                            int* out_type);
 
-
 /*!
-* \brief Update parameters for a Dataset
-* \param handle a instance of data matrix
-* \param parameters parameters
+ * @fn LGBM_DatasetUpdateParam
+ * @brief Update parameters for a Dataset.
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: Decapitalize `for dataset`

File: include/LightGBM/c_api.h [hunk 9]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
+ * @param handle An instance of data matrix
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `An instance of data matrix` -> `An instance of dataset`

File: include/LightGBM/c_api.h [hunk 10]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
 */
 LIGHTGBM_C_EXPORT int LGBM_DatasetUpdateParam(DatasetHandle handle, const char* parameters);
 
 /*!
-* \brief get number of data.
-* \param handle the handle to the dataset
-* \param out The address to hold number of data
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_DatasetGetNumData
+ * @brief Get number of data.
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `number of data` -> `number of data points`?

File: include/LightGBM/c_api.h [hunk 11]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
+ * @param handle The handle to the dataset
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `The handle to the dataset` -> `Handle of dataset`
For the consistency with other descriptions.

File: include/LightGBM/c_api.h [hunk 12]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
+ * @param[out] out The address to hold number of data
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `number of data` -> `number of data points`?

File: include/LightGBM/c_api.h [hunk 13]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
                                              int* out);
 
 /*!
-* \brief get number of features
-* \param handle the handle to the dataset
-* \param out The output of number of features
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_DatasetGetNumFeature
+ * @brief Get number of features.
+ * @param handle The handle to the dataset
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `The handle to the dataset` -> `Handle of dataset`
For the consistency with other descriptions.

File: include/LightGBM/c_api.h [hunk 14]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
+ * @param out The output of number of features
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: ` * @param[out] out The address to hold  number of features`
For the consistency with the function above.

File: include/LightGBM/c_api.h [hunk 15]
```diff
@@ -324,293 +343,333 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetSetField(DatasetHandle handle,
...
 LIGHTGBM_C_EXPORT int LGBM_DatasetGetNumFeature(DatasetHandle handle,
                                                 int* out);
 
 /*!
-* \brief Add features from source to target, then free source
-* \param target The handle of the dataset to add features to
-* \param source The handle of the dataset to take features from
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_DatasetAddFeaturesFrom
+ * @brief Add features from source to target, then free source.
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: Remove `... , then free source`, it's not true anymore. Refer to https://github.com/Microsoft/LightGBM/pull/2006#discussion_r256226978.

File: include/LightGBM/c_api.h [hunk 16]
```diff
@@ -974,3 +1061,5 @@ inline void LGBM_SetLastError(const char* msg) {
 }
 
 #endif  // LIGHTGBM_C_API_H_
+
+
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: Remove these extra lines please, leave only one.

**Comment by guolinke** [comment]
@StrikerRUS yeah, I will take a look for #2059

File: include/LightGBM/c_api.h [hunk 17]
```diff
@@ -854,25 +924,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMats(BoosterHandle handle,
...
 /*!
-* \brief save model into file
-* \param handle handle
-* \param num_iteration, <= 0 means save all
-* \param filename file name
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterSaveModel
+ * @brief Save model into file.
+ * @param handle Handle of booster
+ * @param start_iteration
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Start index of the iteration that should be saved`

File: include/LightGBM/c_api.h [hunk 18]
```diff
@@ -854,25 +924,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMats(BoosterHandle handle,
...
+ * @param num_iteration <= 0 means save all
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Index of the iteration that should be saved, <= 0 means ...`

File: include/LightGBM/c_api.h [hunk 19]
```diff
@@ -854,25 +924,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMats(BoosterHandle handle,
...
-* \param handle handle
-* \param num_iteration, <= 0 means save all
-* \param buffer_len string buffer length, if buffer_len < out_len, re-allocate buffer
-* \param out_len actual output length
-* \param out_str string of model, need to pre-allocate memory before call this
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterSaveModelToString
+ * @brief Save model to string.
+ * @param handle Handle of booster
+ * @param start_iteration
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Start index of the iteration that should be saved`

File: include/LightGBM/c_api.h [hunk 20]
```diff
@@ -854,25 +924,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMats(BoosterHandle handle,
...
+ * @param num_iteration <= 0 means save all
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Index of the iteration that should be saved, <= 0 means ...`

File: include/LightGBM/c_api.h [hunk 21]
```diff
@@ -854,25 +924,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMats(BoosterHandle handle,
...
+ * @param buffer_len String buffer length, if buffer_len < out_len, re-allocate buffer
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `... , you should re-allocate buffer`

File: include/LightGBM/c_api.h [hunk 22]
```diff
@@ -854,25 +924,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMats(BoosterHandle handle,
...
+ * @param[out] out_len Actual output length
+ * @param[out] out_str String of model, need to pre-allocate memory before call this
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `... , should pre-allocate memory`

File: include/LightGBM/c_api.h [hunk 23]
```diff
@@ -882,13 +956,15 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterSaveModelToString(BoosterHandle handle,
...
-* \param handle handle
-* \param num_iteration, <= 0 means save all
-* \param buffer_len string buffer length, if buffer_len < out_len, re-allocate buffer
-* \param out_len actual output length
-* \param out_str json format string of model, need to pre-allocate memory before call this
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterDumpModel
+ * @brief Dump model to JSON.
+ * @param handle Handle of booster
+ * @param start_iteration
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Start index of the iteration that should be dumped`

File: include/LightGBM/c_api.h [hunk 24]
```diff
@@ -882,13 +956,15 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterSaveModelToString(BoosterHandle handle,
...
+ * @param num_iteration <= 0 means save all
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Index of the iteration that should be dumped, <= 0 means ...`

File: include/LightGBM/c_api.h [hunk 25]
```diff
@@ -882,13 +956,15 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterSaveModelToString(BoosterHandle handle,
...
+ * @param buffer_len String buffer length, if buffer_len < out_len, re-allocate buffer
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `... , you should re-allocate buffer`

File: include/LightGBM/c_api.h [hunk 26]
```diff
@@ -882,13 +956,15 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterSaveModelToString(BoosterHandle handle,
...
+ * @param[out] out_len Actual output length
+ * @param[out] out_str JSON format string of model, need to pre-allocate memory before call this
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `... , should pre-allocate memory`

File: include/LightGBM/c_api.h [hunk 27]
```diff
@@ -898,63 +974,78 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterDumpModel(BoosterHandle handle,
...
+ * @fn LGBM_NetworkFree
+ * @brief Finalize the network.
+ * @return 0 when succeed, -1 when failure happens
 */
 LIGHTGBM_C_EXPORT int LGBM_NetworkFree();
 
+
+/*!
+ * @fn LGBM_NetworkInitWithFunctions
+ * @brief
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Initialize the network with external collective functions.`

File: include/LightGBM/c_api.h [hunk 28]
```diff
@@ -898,63 +974,78 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterDumpModel(BoosterHandle handle,
...
+ * @brief Finalize the network.
+ * @return 0 when succeed, -1 when failure happens
 */
 LIGHTGBM_C_EXPORT int LGBM_NetworkFree();
 
+
+/*!
+ * @fn LGBM_NetworkInitWithFunctions
+ * @brief 
+ * @param num_machines
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Total number of machines`

File: include/LightGBM/c_api.h [hunk 29]
```diff
@@ -898,63 +974,78 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterDumpModel(BoosterHandle handle,
...
+ * @param rank
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Rank of local machine`

File: include/LightGBM/c_api.h [hunk 30]
```diff
@@ -898,63 +974,78 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterDumpModel(BoosterHandle handle,
...
+ * @param reduce_scatter_ext_fun
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `The external reduce-scatter function`

File: include/LightGBM/c_api.h [hunk 31]
```diff
@@ -898,63 +974,78 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterDumpModel(BoosterHandle handle,
...
+ * @param allgather_ext_fun
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `The external allgather function`

File: include/LightGBM/c_api.h [hunk 32]
```diff
@@ -639,28 +703,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
...
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterPredictForCSR
+ * @brief Make prediction for an new dataset.
+ *        Note: Should pre-allocate memory for out_result:
+ *              For normal and raw score: its length is equal to num_class * num_data;
+ *              For leaf index, its length is equal to num_class * num_data * num_iteration.
+ * @param handle Handle of booster
+ * @param indptr Pointer to row headers
+ * @param indptr_type Type of indptr, can be C_API_DTYPE_INT32 or C_API_DTYPE_INT64
+ * @param indices Findex
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Pointer to column indices`

File: include/LightGBM/c_api.h [hunk 33]
```diff
@@ -639,28 +703,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
...
+ * @param data Fvalue
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Pointer to the data space`

File: include/LightGBM/c_api.h [hunk 34]
```diff
@@ -639,28 +703,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
...
+ * @param data_type Type of data pointer, can be C_API_DTYPE_FLOAT32 or C_API_DTYPE_FLOAT64
+ * @param nindptr Number of rows in the matrix + 1
+ * @param nelem Number of nonzero elements in the matrix
+ * @param num_col Number of columns; when it's set to 0, then guess from data
+ * @param predict_type
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `* @param predict_type What should be predicted`
and add
```
C_API_PREDICT_CONTRIB: feature contributions (SHAP values)
```

File: include/LightGBM/c_api.h [hunk 35]
```diff
@@ -639,28 +703,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
...
+ *          C_API_PREDICT_NORMAL: normal prediction, with transform (if needed)
+ *          C_API_PREDICT_RAW_SCORE: raw score
+ *          C_API_PREDICT_LEAF_INDEX: leaf index
+ * @param num_iteration Number of iteration for prediction, <= 0 means no limit
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `iteration` -> `iterations`

File: include/LightGBM/c_api.h [hunk 36]
```diff
@@ -639,28 +703,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
...
+ * @param parameter Other parameters for the parameters, e.g. early stopping for prediction.
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `for the parameters` -> `for prediction`
Remove dot at the end.

File: include/LightGBM/c_api.h [hunk 37]
```diff
@@ -639,28 +703,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
...
+ * @param[out] out_len Len of output result
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Len` -> `Length`

File: include/LightGBM/c_api.h [hunk 38]
```diff
@@ -639,28 +703,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
...
+ * @param[out] out_result Used to set a pointer to array, should allocate memory before call this function
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Pointer to array with predictions`

File: include/LightGBM/c_api.h [hunk 39]
```diff
@@ -678,29 +743,30 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSR(BoosterHandle handle,
...
-* \param predict_type
-*          C_API_PREDICT_NORMAL: normal prediction, with transform (if needed)
-*          C_API_PREDICT_RAW_SCORE: raw score
-*          C_API_PREDICT_LEAF_INDEX: leaf index
-* \param num_iteration number of iteration for prediction, <= 0 means no limit
-* \param parameter Other parameters for the parameters, e.g. early stopping for prediction.
-* \param out_len len of output result
-* \param out_result used to set a pointer to array, should allocate memory before call this function
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterPredictForCSRSingleRow
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: The same comments as for `LGBM_BoosterPredictForCSR`

File: include/LightGBM/c_api.h [hunk 40]
```diff
@@ -719,28 +785,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSRSingleRow(BoosterHandle handle,
...
-* \param predict_type
-*          C_API_PREDICT_NORMAL: normal prediction, with transform (if needed)
-*          C_API_PREDICT_RAW_SCORE: raw score
-*          C_API_PREDICT_LEAF_INDEX: leaf index
-* \param num_iteration number of iteration for prediction, <= 0 means no limit
-* \param parameter Other parameters for the parameters, e.g. early stopping for prediction.
-* \param out_len len of output result
-* \param out_result used to set a pointer to array, should allocate memory before call this function
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterPredictForCSC
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: The same comments as for `LGBM_BoosterPredictForCSR`, if other ones are not presented.

File: include/LightGBM/c_api.h [hunk 41]
```diff
@@ -719,28 +785,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSRSingleRow(BoosterHandle handle,
...
+ * @brief Make prediction for an new dataset.
+          Note: Should pre-allocate memory for out_result:
+                For normal and raw score: its length is equal to num_class * num_data;
+                For leaf index, its length is equal to num_class * num_data * num_iteration.
+ * @param handle Handle of booster
+ * @param col_ptr Pointer to col headers
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `col` -`column`

File: include/LightGBM/c_api.h [hunk 42]
```diff
@@ -719,28 +785,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSRSingleRow(BoosterHandle handle,
...
+ * @param col_ptr_type Type of col_ptr, can be C_API_DTYPE_INT32 or C_API_DTYPE_INT64
+ * @param indices Findex
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Pointer to row indices`

File: include/LightGBM/c_api.h [hunk 43]
```diff
@@ -719,28 +785,29 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSRSingleRow(BoosterHandle handle,
...
+ * @param data Fvalue
+ * @param data_type Type of data pointer, can be C_API_DTYPE_FLOAT32 or C_API_DTYPE_FLOAT64
+ * @param ncol_ptr Number of cols in the matrix + 1
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `cols` -> `columns`

File: include/LightGBM/c_api.h [hunk 44]
```diff
@@ -758,25 +825,26 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSC(BoosterHandle handle,
...
-* \param predict_type
-*          C_API_PREDICT_NORMAL: normal prediction, with transform (if needed)
-*          C_API_PREDICT_RAW_SCORE: raw score
-*          C_API_PREDICT_LEAF_INDEX: leaf index
-* \param num_iteration number of iteration for prediction, <= 0 means no limit
-* \param parameter Other parameters for the parameters, e.g. early stopping for prediction.
-* \param out_len len of output result
-* \param out_result used to set a pointer to array, should allocate memory before call this function
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterPredictForMat
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: The same comments as for `LGBM_BoosterPredictForCSR`, if other ones are not presented.

File: include/LightGBM/c_api.h [hunk 45]
```diff
@@ -758,25 +825,26 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSC(BoosterHandle handle,
...
+ * @brief Make prediction for an new dataset.
+          Note: Should pre-allocate memory for out_result:
+                For normal and raw score: its length is equal to num_class * num_data;
+                For leaf index, its length is equal to num_class * num_data * num_iteration.
+ * @param handle Handle of booster
+ * @param data Pointer to the data space
+ * @param data_type Type of data pointer, can be C_API_DTYPE_FLOAT32 or C_API_DTYPE_FLOAT64
+ * @param nrow Number of rows
+ * @param ncol Number columns
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `Number of columns`

File: include/LightGBM/c_api.h [hunk 46]
```diff
@@ -758,25 +825,26 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForCSC(BoosterHandle handle,
...
+ * @param is_row_major 1 for row major, 0 for column major
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `row-major`, `column-major`

File: include/LightGBM/c_api.h [hunk 47]
```diff
@@ -791,27 +859,28 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMat(BoosterHandle handle,
...
-*          C_API_PREDICT_NORMAL: normal prediction, with transform (if needed)
-*          C_API_PREDICT_RAW_SCORE: raw score
-*          C_API_PREDICT_LEAF_INDEX: leaf index
-* \param num_iteration number of iteration for prediction, <= 0 means no limit
-* \param parameter Other parameters for the parameters, e.g. early stopping for prediction.
-* \param out_len len of output result
-* \param out_result used to set a pointer to array, should allocate memory before call this function
-* \return 0 when succeed, -1 when failure happens
-*/LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMatSingleRow(BoosterHandle handle,
+ * @fn LGBM_BoosterPredictForMatSingleRow
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: The same comments as for `LGBM_BoosterPredictForMat`.

File: include/LightGBM/c_api.h [hunk 48]
```diff
@@ -823,24 +892,25 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMat(BoosterHandle handle,
...
-* \param predict_type
-*          C_API_PREDICT_NORMAL: normal prediction, with transform (if needed)
-*          C_API_PREDICT_RAW_SCORE: raw score
-*          C_API_PREDICT_LEAF_INDEX: leaf index
-* \param num_iteration number of iteration for prediction, <= 0 means no limit
-* \param parameter Other parameters for the parameters, e.g. early stopping for prediction.
-* \param out_len len of output result
-* \param out_result used to set a pointer to array, should allocate memory before call this function
-* \return 0 when succeed, -1 when failure happens
+ * @fn LGBM_BoosterPredictForMats
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: The same comments as for `LGBM_BoosterPredictForMat`.

File: include/LightGBM/c_api.h [hunk 49]
```diff
@@ -823,24 +892,25 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterPredictForMat(BoosterHandle handle,
...
+ * @brief Make prediction for an new dataset.
```
Comment (StrikerRUS) [inline include/LightGBM/c_api.h]: `... presented in a form of array of pointers to rows.`
Refer to https://github.com/Microsoft/LightGBM/pull/2008#issue-252504393.

**Comment by guolinke** [comment]
@StrikerRUS maybe you can fix these minor issues directly and then this PR could be merged.

**Comment by StrikerRUS** [comment]
@guolinke Sure, I think I can take this PR over in case of author's further inactivity.

**Comment by StrikerRUS** [comment]
@guolinke BTW, while we are speaking about abandoned PRs, can you please share your opinion in #1864?

**Comment by StrikerRUS** [comment]
ping @nabokovas 
Are you with us?

**Comment by nabokovas** [comment]
I am sorry for the really long delay

**Comment by StrikerRUS** [comment]
@nabokovas No problem!
Thanks a lot for your patience and valuable contribution!

**Pull request merged**